### PR TITLE
Checks for null byte injection have been added

### PR DESCRIPTION
**Pull request by masdevas**

### CODE REVIEW

File: include/data_management/data_source/mysql_feature_manager.h [hunk 1]
```diff
@@ -304,10 +314,10 @@ class SQLFeatureManager
         return status;
     }
 
-private:
+public:
     internal::SQLFetchBufferPtr _fetchBuffer;
     BlockDescriptor<DAAL_DATA_TYPE> _currentRowBlock;
-    services::SharedPtr<services::ErrorCollection> _errors;
+    services::SharedPtr<services::ErrorCollection> _errors = services::SharedPtr<services::ErrorCollection>(new services::ErrorCollection);
```
Comment (SmirnovEgorRu) [inline include/data_management/data_source/mysql_feature_manager.h]: This works starting from C++11 only. In header files DAAL supports C++ starting from C++03 version. So, you should replace this.

File: include/data_management/data_source/mysql_feature_manager.h [hunk 2]
```diff
@@ -304,10 +314,10 @@ class SQLFeatureManager
         return status;
     }
 
-private:
+public:
```
Comment (SmirnovEgorRu) [inline include/data_management/data_source/mysql_feature_manager.h]: What is the reason for?

File: include/services/error_indexes.h [hunk 3]
```diff
@@ -389,6 +389,7 @@ enum ErrorID
     ErrorSQLstmtHandle = -90044,                                        /*!< ErrorSQLstmtHandle */
     ErrorOnFileOpen = -90045,                                           /*!< Error on file open */
     ErrorOnFileRead = -90046,                                           /*!< Error on file read */
+    ErrorNullByteInjection = -90047,
```
Comment (SmirnovEgorRu) [inline include/services/error_indexes.h]: You also should add description for the error in file services/kernel/error_handling.cpp

File: include/services/error_indexes.h [hunk 4]
```diff
@@ -389,6 +389,7 @@ enum ErrorID
     ErrorSQLstmtHandle = -90044,                                        /*!< ErrorSQLstmtHandle */
     ErrorOnFileOpen = -90045,                                           /*!< Error on file open */
     ErrorOnFileRead = -90046,                                           /*!< Error on file read */
+    ErrorNullByteInjection = -90047,                                    /*!< Error null byte injection */
```
Comment (SmirnovEgorRu) [inline include/services/error_indexes.h]: The comment is still should be applied: you should add description for the error in file "services/kernel/error_handling.cpp"

**Comment by SmirnovEgorRu** [comment]
make test

**Comment by SmirnovEgorRu** [comment]
Windows examples in dynamic linkage are crushed with:
unresolved external symbol "class daal::services::interface1::Status __cdecl daal::services::internal::checkForNullByteInjection(char const *,char const * const)

Should be addressed.

**Comment by masdevas** [comment]
make test

File: service/kernel/error_handling.cpp [hunk 5]
```diff
@@ -39,20 +39,6 @@ namespace services
     this->_errors->add(services::ErrorIncorrectNumberOfElementsInResultCollection);
 */
 
-namespace internal {
-Status checkForNullByteInjection(const char *begin, const char *const end)
```
Comment (SmirnovEgorRu) [inline service/kernel/error_handling.cpp]: FYI
Ideally it should be solved by adding DAAL_EXPORT define to the function in .cpp file, moving implementation to header file is not the best way.

**Pull request merged**